src/comments/CommentItem.js

```diff
@@ -60,6 +60,11 @@ export default class CommentItem extends Component {
     }
   }
 
+  updateCommentPostion = () => {
+    // should happen after react-router updated the route
+    setTimeout(() => this.checkHashLink());
```

Comment:
@withRouter is already in the component and first i tried with `componentDidUpdate` new props it was like this 
```js 
componentDidUpdate(prevProps) {
    const prevHash = _.get(prevProps, 'location.hash');
    const currentHash = _.get(this.props, 'location.hash');
    if (prevHash !== currentHash) {
      this.checkHashLink();
    }
  }
```
But router was not taking new position I think that has to do with `react-router-scroll/lib/useScroll`